airflow/providers/google/cloud/hooks/bigquery_reservation.py

```diff
@@ -0,0 +1,129 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""This module contains a BigQuery Reservation Hook."""
+from __future__ import annotations
+
+from typing import Sequence
+
+from google.cloud.bigquery_reservation_v1 import ReservationServiceClient
+
+from airflow.exceptions import AirflowException
+from airflow.providers.google.common.consts import CLIENT_INFO
+from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
+
+
+class BigQueryReservationServiceHook(GoogleBaseHook):
+    """
+    Hook for Google Bigquery Reservation API.
+
+    :param gcp_conn_id: The Airflow connection used for GCP credentials.
+    :param location: The location of the BigQuery resource.
+    :param delegate_to: This performs a task on one host with reference to other hosts.
+    :param impersonation_chain: This is the optional service account to impersonate using short term
+        credentials.
+    """
+
+    conn_name_attr = "gcp_conn_id"
+    default_conn_name = "google_cloud_bigquery_reservation_default"
+    conn_type = "gcpbigqueryreservation"
+    hook_name = "Google Bigquery Reservation"
+
+    def __init__(
+        self,
+        gcp_conn_id: str = GoogleBaseHook.default_conn_name,
+        location: str | None = None,
+        delegate_to: str | None = None,
```

Comment:
in `GoogleBaseHook` we have deprecated `delegate_to` so can we remove this from here?